config/routes.rb

```diff
@@ -103,7 +103,12 @@
           post "save_status"
         end
       end
-      resources :tags, only: %i[index update show]
+      resources :tags, only: %i[index update show] do
+        member do
```

Comment:
I know we don't do this elsewhere, but I really prefer routing like this:

```ruby
  resource :tag_admin, only: %i[create destroy] # or update if you really want patch
```

Then you make a separate nested controller with two actions and everything is cleanly separated and the routes are actually ResSTful.

http://jeromedalbert.com/how-dhh-organizes-his-rails-controllers/

Reply:
Hmm so I tried:
```ruby
namespace: admin do
  resources :tags, only: %i[index update show] do
    resource :tag_moderator, only: %i[create destroy], module: "tags"
  end
end
```

The path is correct, but the controller it hits isn't:
```
# show-routes in console

POST (/locale/:locale)/admin/tags/:tag_id/tag_moderator(.:format)
admin/tags/tag_moderators#create {:locale=>nil}
DELETE (/locale/:locale)/admin/tags/:tag_id/tag_moderator(.:format)
admin/tags/tag_moderators#destroy {:locale=>nil}
```

My controller is named like the example:
```
module Admin
  class Tags::TagModeratorsController < ApplicationController
  end
end
```

Any ideas?

Reply:
@Zhao-Andy it looks like the right controller. That's where you placed your controller correct? also, since it's already in a module I would think `Tags::ModeratorsController` is better, you can ommit the 2nd `Tag`.

Reply:
Yeah, agree that moderators alone would look better. The above code doesn't work though, I keep getting `Uninitialized constant Admin::Tags`.

Reply:
@Zhao-Andy The route is correct, you just don't have that controller. What I meant (taking @maestromac's sensible suggestion re naming into account):
`config/routes.rb`
```ruby
namespace: admin do
  resources :tags, only: %i[index update show] do
    resource :moderator, only: %i[create destroy], module: "tags"
  end
end
```
`app/controllers/admin/tags/moderator_controller.rb`
```ruby
module Admin
  module Tags
    module ModeratorsController < ApplicationController
    
      def create
         ...
      end

      def destroy
        ...
      end
    end
  end
end
```

Reply:
Ahh thanks! My file path was the wrong part -- I had it only under `/admin`.